docs/guide.md

```diff
@@ -0,0 +1,367 @@
+
+[design]: ../DESIGN.md
+
+
+MySQL setup
+-----------
+
+Fleetspeak uses MySQL as its primary data store. Make sure that you have a
+relatively new MySQL or MariaDB installation—please refer to the official
+documentation for instructions specific to your platform.
+
+Fire-up the MySQL console as an administrative user, e.g.:
+
+    $ mysql --user root
+
+Create a Fleetspeak database and an associated database user:
+
+    mysql> CREATE USER `fleetspeak-user` IDENTIFIED BY 'fleetspeak-password';
+    mysql> CREATE DATABASE `fleetspeak`;
+    mysql> GRANT ALL PRIVILEGES ON `fleetspeak`.* TO `fleetspeak-user`;
+
+Remember the database name, the username and the password—they will be specified
+later in the Fleetspeak configuration file.
+
+
+Fleetspeak compilation
+----------------------
+
+Fleetspeak is written in the [Go programming language][golang]. Therefore, in
+order to compile it you need to install the Go development environment on your
+system. As always, refer to the official documentation on how to do it (or use
+your platform's package manager).
+
+In order to simply compile everything, setup `$GOPATH` to a folder of your
+choice (it will be used by the Go to store the sources and resulting binary
+files) and run the `go get` tool:
+
+    $ export GOPATH="$HOME/fleetspeak"
+    $ go get github.com/google/fleetspeak/...
+
+That's it! Now, in your `$GOPATH/bin` you should have multiple Fleetspeak binary
+files available: `admin`, `client`, `config`, `server`.
+
+
+[golang]: https://golang.org/
+
+
+Fleetspeak configuration
+------------------------
+
+Fleetspeak uses a bunch of configuration files written in the
+[Protocol Buffers][protobuf] text format. Thanks to this, you can always inspect
+the proto definition to learn about the schema when in doubt.
+
+We will start by creating a basic configuration file for the Fleetspeak
+configurator—a binary that generates more specific configuration files,
+certificates and keys.
+
+Save the following in some file, e.g. `$HOME/.config/fleetspeak.textproto`:
+
+```
+configuration_name: "Example"
+
+components_config {
+
+  mysql_data_source_name: "fleetspeak-user:fleetspeak-password@/fleetspeak"
+
+  https_config {
+    listen_address: "localhost:9090"
+  }
+
+  admin_config {
+    listen_address: "localhost:9091"
+  }
+}
+
+public_host_port: "localhost:9090"
+
+trusted_cert_file: "$HOME/.config/fleetspeak-server/ca.pem"
+trusted_cert_key_file: "$HOME/.config/fleetspeak-server/ca-key.pem"
+
+server_cert_file: "$HOME/.config/fleetspeak-server/server.pem"
+server_cert_key_file: "$HOME/.config/fleetspeak-server/server-key.pem"
+
+server_component_configuration_file: "$HOME/.config/fleetspeak-server/components.textproto"
+linux_client_configuration_file: "$HOME/.config/fleetspeak-client/config.textproto"
+```
+
+Refer to the schema for details about each of these preferences and other
+possible options.
+
+Note that the configurator is currently not able to expand variables. If you
```

Comment:
I think it'd be simpler if the doc suggests to create files in /tmp/fs-config, for example. Then there's no need to use $HOME in the proto file and then explain later that it won't get expanded and must be expanded manually. Fundamentally, people tend to copy&paste from the docs, so using $HOME is likely to be an issue.

Reply:
Actually, inside the configurator config, I'd just use /etc/fleetspeak-server or the likes.

Reply:
I agree that usage of `$HOME` in the examples can be confusing, so I have changed the guide to use a sample path and explained that it should be changed accordingly.

Using `/tmp` is a very poor choice for this, because there are no guarantees about how long the files there are preserved (e.g. usually, this is wiped on every system boot). I would not like to follow some guide, take a break and have all my work go to waste when I return later.

Likewise, I have purposefully avoided using `/etc`. I don't like polluting my global system configuration with toy setups. Moreover, in many cases that would require super-user privileges, which may be very disheartening (I know it would definitely be for me).